lib/Parse/ParseExpr.cpp

```diff
@@ -2109,7 +2109,7 @@ void Parser::parseOptionalArgumentLabel(Identifier &name, SourceLoc &loc) {
     // the syntax for referring to the function pointer (foo(_:)),
     auto escaped = Tok.isEscapedIdentifier();
     auto underscore = Tok.is(tok::kw__) || (escaped && text == "_");
-    if (escaped && !underscore && canBeArgumentLabel(text)) {
+    if (escaped && !underscore && shouldBeArgumentLabel(text)) {
```

Comment:
I think it's okay to use `!Tok.isAny(tok::kw_let, tok::kw_var)` here instead of adding `shouldBeArgumentLabel` which sounds kind of out of place to me...

Reply:
That was my initial instinct as well but unfortunately that won't work as the token being processed here isn't `tok::kw_let` or `tok::kw_var`. At this point, it's `tok::identifier` with the text "let" or "var". That's why the initial implementation called `StringExtras::canBeArgumentLabel` rather than `Token::canBeArgumentLabel`.

If you think it's cleaner to do the string comparison inline rather than adding shouldBeArgumentLabel, I can do that, but I found that a bit hard to read.

Reply:
Ah interesting, let's convert that method into an inline closure then with `requiresEscaping` name or something similar.

Reply:
Just to clarify, do you mean something like this?
```
auto requiresEscaping = [](StringRef identifier) {
  return llvm::StringSwitch<bool>(identifier)
    .Case("let", false)
    .Case("var", false)
    .Default(canBeArgumentLabel(identifier));
};

if (escaped && !underscore && requiresEscaping(text)) {
  ...
}
```

Reply:
Yeah, that's exactly what I mean.

Reply:
👍